crypto/pem/pem_lib.c

```diff
@@ -954,6 +954,11 @@ int PEM_read_bio_ex(BIO *bp, char **name_out, char **header,
     EVP_DecodeInit(ctx);
     BIO_get_mem_ptr(dataB, &buf_mem);
     len = buf_mem->length;
+
+    /* There was no data in the PEM file */
+    if (len == 0)
+        goto end;
+
```

Comment:
If you move down the EVP_ENCODE_CTX_new() call, you can have this length check even earlier...

Reply:
Fixed